File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1961_basico.js
import { question } from "readline-sync";

function main() {
    let saltoCanos = get_Text().split(` `).map(Number);
    let alturaCanos = get_Text().split(` `).map(Number);
    let vitoria = true;

    for (let i = 0; i < saltoCanos[1] - 1; i++) {
        if (Math.abs(alturaCanos[i] - alturaCanos[i + 1]) > saltoCanos[0]) {
            vitoria = false;
            break;
        }
    }

    if (vitoria === true) {
        print(`YOU WIN`);
    } else {
        print(`GAME OVER`);
    }

}

function get_Text() {
    return question();
}

function print(texto) {
    return console.log(texto);
}

main();

File: Exercicios_ED_I/Bee/1961_basico.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
#include <sstream>
#include <math.h>
using namespace std;

main() {
    int alturaPulo, numCanos, condicao = 1;
    vector<int> alturasCanos;
    cin >> alturaPulo >> numCanos;
    cin.ignore();
    
    string alturaCanos;
    getline(cin, alturaCanos);
    istringstream stream (alturaCanos);
    int numeroAlturaCano;

    while (stream >> numeroAlturaCano) {
        alturasCanos.push_back(numeroAlturaCano);
    }

    size_t i;

    for (i = 0; i < alturasCanos.size() - 1; i++) {
        if (abs(alturasCanos[i] - alturasCanos[i + 1]) > alturaPulo) {
            condicao = 0;
            break;
        }
    }
    
    if (condicao) {
        cout << "YOU WIN" << endl;
    } else {
        cout << "GAME OVER" << endl;
    }
}